iam-login-service/src/main/java/it/infn/mw/iam/api/aup/AupSignaturePageController.java

```diff
@@ -128,6 +135,34 @@ public ModelAndView signAup(HttpServletRequest request, HttpServletResponse resp
 
     return new ModelAndView("redirect:/dashboard");
   }
+
+  @PreAuthorize("hasRole('USER')")
+  @RequestMapping(method = RequestMethod.PATCH, value = "/iam/aup/sign")
+  public void resignAup(HttpServletRequest request, HttpServletResponse response) {
+
+    Optional<IamAup> aup = repo.findDefaultAup();
```

Comment:
`setSignatureForAccount` is using ` signatureRepo.createSignatureForAccount(account, dto.getSignatureTime());`. Instead of updating the time it is using the existing time from the signature. If I modify the method with `signatureRepo.updateSignatureForAccount(account, now)`,  test `updateAupWithRightScope` is failing because in case of `createSignatureForAccount` if there is no signature it creates a new one.
Can you please confirm the we don't need the functionality of creating a signature in PATCH operation, in that case I could modify the test as well.

Reply:
Yes, my comment was originally something like "this PATCH method should be moved inside another controller or another method could be fixed and used" but you're right and PATCH method seems not necessary in the end, due to the implementation of createSignatureForAccount that creates if missing and replaces signature if it exists. I add that this `updateSignatureForAccount` method, seeing its implementation, seems also useless and probably can be removed.

I suggest to use the signature API, in particular I see that the [POST](https://github.com/indigo-iam/iam/blob/master/iam-persistence/src/main/java/it/infn/mw/iam/persistence/repository/IamAupSignatureRepositoryImpl.java#L58) is (wrongly?!) implemented in a way that if signature exists the signature time is updated. Then, can we simply use this API call? Probably yes. This call can be accessed by authorized users to (re)sign their AUP, so there's no authz problems to be fixed.